lua/lspconfig/dartls.lua

```diff
@@ -5,7 +5,10 @@ local server_name = 'dartls'
 local bin_name = 'dart'
 
 local find_dart_sdk_root_path = function()
-  if vim.fn['executable'] 'flutter' == 1 then
+  if os.getenv 'FLUTTER_SDK' ~= nil then
+    local flutter_path = os.getenv 'FLUTTER_SDK'
+    return flutter_path .. '/cache/dart-sdk/bin/dart'
```

Comment:
this is not windows compatible, we have some path utils for joining

Reply:
I was going off line 14, where it also concatenates a string to append to the path.

Reply:
Yes, I didn't write the original implementation so I just see this as an opportunity for some cleanup :)

Reply:
Okay, working on it now

Reply:
https://github.com/neovim/nvim-lspconfig/blob/4f72377143fc0961391fb0e42e751b9f677fca4e/lua/lspconfig/util.lua#L140-L143

Reply:
that's not quite it :) You need to separate out the directories into a table.